File: CCI/ArraysAndStringsC1/1_1_IsUnique/IsUnique.js

function isUnique(text){
    let characters = [];
    let isUniqueChar = true;

    for(let char of text){
        if(characters.includes(char)){
            isUniqueChar = false;
            break;
        }

        characters.push(char);
    }

    return isUniqueChar;

}

const inputs = [
    'Hello, This is the first test',
    '#21ds2fhgQsq',
    'My other',
    'JEUSIiasi',
    'Unique char'
];


for(let i of inputs){
    result = isUnique(i);
    resultText = `
    Result
        Input: ${i}
        Output: ${result}
    `;
    console.log(resultText);
}

File: CCI/ArraysAndStringsC1/1_1_IsUnique/IsUnique.py

def is_unique(text):
    character_dict = []
    is_unique_char = True

    for character in text:
        if character in character_dict:
            is_unique_char = False
            break
        character_dict.append(character)
    
    return is_unique_char




if __name__ == '__main__':
    inputs = [
        'Hello, This is the first test',
        '#21ds2fhgQsq',
        'My other',
        'JEUSIiasi',
        'Unique char',
    ]

    for i in inputs:
        result = is_unique(i)
        result_print = f"""
        Result
            Input: {i}
            Output: {result}
        """
        print(result_print)
